### PR TITLE
Added conditional function and ability to output additional data

### PR DESCRIPTION
**Pull request by kkctor**
Added a conditional function that allows one to control the amount of output received.
Also added the ability to pass in additional data to be printed - this is mostly useful with `timeit_section` and is pretty much irrelevant with `timeit_function`.

### CODE REVIEW

**Comment by narekgharibyan** [comment]
@kkctor can you please have a look into failing CI ?

File: ptimeit/_impl.py [hunk 1]
```diff
@@ -34,18 +42,31 @@ def timeit_section(section_name: str):
...
-        print(f'->>>>>>>>{elapsed_time_ms_str}ms      {section_full_name}')
+        if should_print is None or should_print(elapsed_time_ms):
+            elapsed_time_ms_str = str(elapsed_time_ms).rjust(14)
+            offset = '|   ' * _depth
+            section_full_name = offset + section_name
+            print(f'->>>>>>>>{elapsed_time_ms_str}ms      {section_full_name}')
+            if additional_data_to_print:
+                print(f'->>>>>>>>>>>>>>>>>>>>>>>>      {offset}'
+                f'Additional data: {additional_data_to_print}'
+            )
```
Comment (narekgharibyan) [inline ptimeit/_impl.py]: Maybe it's better to print this on the same line with the rest of info ? 🤔 
Just am not sure where and how this is intended to be used.

File: ptimeit/_impl.py [hunk 2]
```diff
@@ -34,18 +42,31 @@ def timeit_section(section_name: str):
     finally:
         elapsed_time_s = time.monotonic() - start_time
         elapsed_time_ms = round(float(elapsed_time_s * 1000), 1)
-        elapsed_time_ms_str = str(elapsed_time_ms).rjust(14)
-        section_full_name = '|   ' * _depth + section_name
-        print(f'->>>>>>>>{elapsed_time_ms_str}ms      {section_full_name}')
+        if should_print is None or should_print(elapsed_time_ms):
```
Comment (narekgharibyan) [inline ptimeit/_impl.py]: As we already have default value, prob we can drop the `should_print is None` check.

File: ptimeit/_impl.py [hunk 3]
```diff
@@ -1,17 +1,25 @@
...
 
 
 _depth = -1
 
 
 @contextmanager
-def timeit_section(section_name: str):
+def timeit_section(
+    section_name: str,
+    should_print: Callable[[float], bool] = lambda t: True,
```
Comment (narekgharibyan) [inline ptimeit/_impl.py]: maybe we can just name this `cond` or `condition` ?

**Review by narekgharibyan — CHANGES_REQUESTED** [review]
@kkctor thanks for you contribution. looks good, just added a few suggestions.

Also CI fails, would be nice if you can check that out.

File: ptimeit/_impl.py (hunk 1 above)
Comment (kkctor) [reply ptimeit/_impl.py]: Alright: https://github.com/narekgharibyan/ptimeit/pull/1/commits/3b0982515d4e9a8b7f2ad5fa736a70b687c688a2

File: ptimeit/_impl.py (hunk 2 above)
Comment (kkctor) [reply ptimeit/_impl.py]: Decided against it and just set the default value to `None`.
If that check is removed and someone manually passes `None` the code will crash.

File: ptimeit/_impl.py (hunk 3 above)
Comment (kkctor) [reply ptimeit/_impl.py]: Yeah: https://github.com/narekgharibyan/ptimeit/pull/1/commits/3b0982515d4e9a8b7f2ad5fa736a70b687c688a2

**Comment by kkctor** [comment]
> @kkctor can you please have a look into failing CI ?

That is fixed in the latest commits. 
It was just an indentation issue that I didn't notice.
You will need to run the workflow to let me know if there's any other similar place that needs to be fixed

**Comment by narekgharibyan** [comment]
Yeah, there is this error:
```
./ptimeit/_impl.py:49:21: E225 missing whitespace around operator

```

Btw, you should be able to run those checks locally too, check out [these](https://github.com/narekgharibyan/ptimeit/blob/master/.github/workflows/build.yml#L27-L30) steps from CI.

File: ptimeit/_impl.py (hunk 2 above)
Comment (narekgharibyan) [reply ptimeit/_impl.py]: Okie

File: ptimeit/_impl.py [hunk 4]
```diff
@@ -1,17 +1,25 @@
...
 
 _depth = -1
 
 
 @contextmanager
-def timeit_section(section_name: str):
+def timeit_section(
+    section_name: str,
+    condition: Callable[[float], bool] = None,
+    additional_data_to_print: str = None
```
Comment (narekgharibyan) [inline ptimeit/_impl.py]: Maybe ? 
```suggestion
    extra_data_to_print: str = None
```

File: ptimeit/_impl.py [hunk 5]
```diff
@@ -34,18 +42,31 @@ def timeit_section(section_name: str):
...
-        elapsed_time_ms_str = str(elapsed_time_ms).rjust(14)
-        section_full_name = '|   ' * _depth + section_name
-        print(f'->>>>>>>>{elapsed_time_ms_str}ms      {section_full_name}')
+        if condition is None or condition(elapsed_time_ms):
+            elapsed_time_ms_str = str(elapsed_time_ms).rjust(14)
+            offset = '|   ' * _depth
+            section_full_name = offset + section_name
+            to_print=f'->>>>>>>>{elapsed_time_ms_str}ms      {section_full_name}'
+            if additional_data_to_print:
+                to_print += f'- {additional_data_to_print}'
```
Comment (narekgharibyan) [inline ptimeit/_impl.py]: Hmm, feels like we need an extra space here or it
's handled somewhere else ? 🤔 
```suggestion
                to_print += f' - {additional_data_to_print}'
```

**Review by narekgharibyan — CHANGES_REQUESTED** [review]
@kkctor would be nice to add a test case for these changes.

File: ptimeit/_impl.py (hunk 4 above)
Comment (kkctor) [reply ptimeit/_impl.py]: https://github.com/narekgharibyan/ptimeit/pull/1/commits/7f0ecc23ff37274d98456cbbae9c847c9bebd38b

File: ptimeit/_impl.py (hunk 5 above)
Comment (kkctor) [reply ptimeit/_impl.py]: https://github.com/narekgharibyan/ptimeit/pull/1/commits/d3ff11903ec596d04d4eafe52b02d500a48bf187

**Comment by kkctor** [comment]
> Yeah, there is this error:
> 
> ```
> ./ptimeit/_impl.py:49:21: E225 missing whitespace around operator
> ```
> 
> Btw, you should be able to run those checks locally too, check out [these](https://github.com/narekgharibyan/ptimeit/blob/master/.github/workflows/build.yml#L27-L30) steps from CI.

Need to do `isort . -rc --check-only` doesn't find anything without `-rc`.

(I have an older version of isort, though)

**Comment by kkctor** [comment]
> @kkctor would be nice to add a test case for these changes.

https://github.com/narekgharibyan/ptimeit/pull/1/commits/bdeb02a65709e4e4568bdfb8f1c10322c6b9cd5a

**Review by narekgharibyan — APPROVED** [review]

**Comment by narekgharibyan** [comment]
@kkctor thanks for your contribution. 
This is already published in [v0.1.2](https://github.com/narekgharibyan/ptimeit/releases/tag/v0.1.2) release.

**Pull request merged**